lib/src/debian.dart

```diff
@@ -0,0 +1,282 @@
+  await Directory('$debianDir/DEBIAN').create(recursive: true);
+  await Directory('$debianDir/usr/local/bin').create(recursive: true);
+  return debianDir;
+}
+
+/// Copy all executable files listed in the map [executables] from the `build` folder
+void _copyExecutableFiles(String debianDir) {
+  final executablePath = p.join(debianDir, "usr", "local", "bin");
+  executables.value.forEach((name, path) {
+    run("cp", arguments: [
+      p.join("build", "$name.native"),
+      p.join(executablePath, name)
+    ]);
+  });
+}
+
+/// Generate the control file for the Debian package.
+void _generateControlFile(String debianDir) {
+  var controlFilePath = p.join(debianDir, "DEBIAN", "control");
+
+  String _updatedControlData = replaceFirstMappedMandatory(
+      controlData.value,
+      RegExp(r'Version: ([0-9].*)'),
+      (match) => 'Version: ${version.toString()}',
+      "Couldn't find a version field in the given CONTROL file.");
+
+  writeString(controlFilePath, _updatedControlData);
+}
+
+/// Scan for new .deb packages in the [repo] and update the `Packages` file.
+Future<void> _updatePackagesFile(String repo) async {
+  // Scan for new packages
+  String output = run("dpkg-scanpackages",
+      arguments: ["--multiversion", "."], workingDirectory: repo);
+  // Write the stdout to the file
+  writeString(p.join(repo, 'Packages'), output);
+  // Force Compress the Packages file
+  run("gzip", arguments: ["-k", "-f", "Packages"], workingDirectory: repo);
+}
+
+/// Generate the Release index for the PPA.
+Future<void> _updateReleaseFile(String repo) async {
+  String output = run("apt-ftparchive",
+      arguments: ["release", "."], workingDirectory: repo);
+  writeString(p.join(repo, 'Release'), output);
+}
+
+/// Sign the Release file with the GPG key.
+Future<void> _updateReleaseGPGFile(String repo) async {
+  run("gpg",
+      arguments: [
+        for (String arg in gpgArgs) arg,
+        "--default-key",
+        (gpgFingerprint.value),
+        "--passphrase",
+        (gpgPassphrase.value),
+        "-abs",
+        "-o",
+        "Release.gpg",
+        "Release",
+      ],
+      quiet: true,
+      workingDirectory: repo);
+}
+
+/// Update the InRelease file with the new index and keys.
+Future<void> _updateInReleaseFile(String repo) async {
+  run("gpg",
+      arguments: [
+        for (String arg in gpgArgs) arg,
+        "--default-key",
+        (gpgFingerprint.value),
+        "--passphrase",
+        (gpgPassphrase.value),
+        "--clearsign",
+        "-o",
+        "InRelease",
+        "Release",
+      ],
+      quiet: true,
+      workingDirectory: repo);
+}
+
+/// Import the private key into the GPG.
+void _importGPGPrivateKey() {
+  log("Importing the GPG Private Key");
+  writeString("._privateKey", gpgPrivateKey.value);
+  run(
+    "gpg",
+    arguments: [
+      for (String arg in gpgArgs) arg,
+      "--passphrase",
+      gpgPassphrase.value,
+      "--allow-secret-key-import",
```

Comment:
According to the gpg manpage, "this is an obsolete option and is not used anywhere".

Reply:
Didn't notice. Thanks for the catch.